### PR TITLE
feat: 'kpm run' supports compile the kcl package from oci registry.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #46 

#### 2. What is the scope of this PR (e.g. component or file name):

kpm/pkg/cmd
kpm/pkg/oci

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [x] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

 'kpm run' supports compile the kcl package from oci registry.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/cmd/cmd_run.go [hunk 1]
```diff
@@ -10,30 +10,31 @@ import (
...
-				Name:  "tar",
-				Usage: "The kcl package tar will be executed",
-			},
 			// The entry kcl file.
 			&cli.StringFlag{
 				Name:  "input",
 				Usage: "a kcl file as the compile entry file",
 			},
+			&cli.StringFlag{
+				Name:  "tag",
```
Comment (Peefy) [inline pkg/cmd/cmd_run.go:35]: Why called `tag` instead of `oci`.

File: pkg/cmd/cmd_run.go [hunk 2]
```diff
@@ -10,30 +10,31 @@ import (
...
+				Usage: "the tag for oci repo",
```
Comment (Peefy) [inline pkg/cmd/cmd_run.go:36]: s/oci repo/oci artifact?

File: pkg/cmd/cmd_run.go [hunk 3]
```diff
@@ -43,53 +44,24 @@ func NewRunCmd() *cli.Command {
...
 
-			tarPath := c.String("tar")
-			compileResult := ""
-
-			// If a tar package specified by "--tar" to run
-			if len(tarPath) != 0 {
-				absTarPath, err := absTarPath(tarPath)
+			// 'kpm run' compile the current package undor '$pwd'.
+			if len(pkgWillBeCompiled) == 0 {
+				compileResult, err := runPkg(c.String("input"), c.Bool("vendor"))
```
Comment (Peefy) [inline pkg/cmd/cmd_run.go:51]: Use constants for flags `input` and `vendor`

File: pkg/cmd/cmd_run.go (hunk 1 above)
Comment (zong-zhe) [reply pkg/cmd/cmd_run.go]: The `tag` is not used to instead `oci`, I have changed the `kpm run -- oci` mode to `kpm run <oci_url>`, `tag` is used to describe the version in `kpm run --tag v0.0.1 <oci_url>`

File: pkg/cmd/cmd_run.go (hunk 2 above)
Comment (zong-zhe) [reply pkg/cmd/cmd_run.go]: fixed.

File: pkg/cmd/cmd_run.go (hunk 3 above)
Comment (zong-zhe) [reply pkg/cmd/cmd_run.go]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**